proxy_session.go

```
@@ -47,6 +47,7 @@ type MetricsHookFactory interface {
 
 type ResponseInterceptor interface {
 	InterceptMongoToClient(m Message, serverAddress address.Address, isRemote bool) (Message, error)
+	ExtractExecutionTime(startTime time.Time, pausedExecutionTime int64)
```

went back and forth on whether this should return an error. i decided **not** to return an error for the following reason: 
* in the proxy code, if we go to set latency stats with an invalid readwritetype i believe there should be an error. however, i don't think this error needs to be surfaced to mongonet. We should not return an error to the user in this situation given it would be an internal, implementation error. Instead, i think the error should be logged within the proxy? 

let me know what you think!

---

agreed - we generally do not return an error to the user on non-blocking internal implementation failures

---

reviewing the name again - could you add a comment clarifying what's expected from the implementation?

---

yeah, i am struggling to think of the most descriptive name for this function. adding a comment here to clarify but open to other naming suggestions too

---

I like the comment, but added a renaming suggestion